src/4626/Factory4626.sol

```diff
@@ -0,0 +1,89 @@
+pragma solidity 0.8.19;
+
+import {ERC20} from "solmate/tokens/ERC20.sol";
+
+import {MErc20} from "@protocol/MErc20.sol";
+import {MoonwellERC4626} from "@protocol/4626/MoonwellERC4626.sol";
+import {Comptroller as IMoontroller} from "@protocol/Comptroller.sol";
+
+contract Factory4626 {
+    /// ------------------------------------------------
+    /// ------------------------------------------------
+    /// ------------------ IMMUTABLES ------------------
+    /// ------------------------------------------------
+    /// ------------------------------------------------
+
+    /// @notice The Moonwell moontroller contract
+    IMoontroller public immutable moontroller;
+
+    /// @notice The WETH9 contract
+    address public immutable weth;
+
+    /// @notice event emitted when a new 4626 vault is deployed
+    /// @param asset underlying the vault
+    /// @param mToken the mToken contract
+    /// @param rewardRecipient the address to receive rewards
+    /// @param deployed the address of the deployed contract
+    event DeployedMoonwellERC4626(
+        address indexed asset,
+        address indexed mToken,
+        address indexed rewardRecipient,
+        address deployed
+    );
+
+    /// @param _moontroller The Moonwell comptroller contract
+    /// @param _weth The WETH9 contract
+    constructor(IMoontroller _moontroller, address _weth) {
+        moontroller = _moontroller;
+        weth = _weth;
+    }
+
+    /// @notice Deploy a MoonwellERC4626 vault
+    /// @param mToken The corresponding mToken
+    /// @param rewardRecipient The address to receive rewards
+    function deployMoonwellERC4626(
+        address mToken,
+        address rewardRecipient
+    ) external returns (address vault) {
+        address asset = MErc20(mToken).underlying();
+        /// parameter checks
+        require(rewardRecipient != address(0), "INVALID_RECIPIENT");
+        require(asset != weth, "INVALID_ASSET");
+
+        /// create the vault contract
+        vault = address(
+            new MoonwellERC4626(
+                ERC20(asset),
+                MErc20(mToken),
+                rewardRecipient,
+                moontroller
+            )
+        );
+
+        /// handle initial mints to the vault to prevent front-running
+        /// and share price manipulation
+
+        uint256 initialMintAmount = 10 ** ((ERC20(asset).decimals() * 2) / 3);
```

Comment:
I calculate that for USDC this would be 10 * ((12)/3) or 40, which would be equivalent to $0.000040

For a 1e18 like WETH it would be 10 * ((36)/3) or 120, which would be equivalent to 0.000000000000000120

If the vault tokens are 1e18 this seems totally fine. I'm unclear on whether ERC4626 vaults suffer from the same share price manipulation attacks that money markets do though, because there is no collateral status or borrowing possible. Do you have more details on this?